src/main/java/edu/ie3/datamodel/utils/validation/GridContainerValidationUtils.java

```diff
@@ -42,10 +41,10 @@ private GridContainerValidationUtils() {
       return List.of(isNull);
     }
 
-    /* sanity check to ensure distinct UUIDs */
-    List<Try<Void, ? extends ValidationException>> exceptions =
-        new ArrayList<>(
-            checkForDuplicates(gridContainer.allEntitiesAsList(), UniqueEntity::getUuid));
+    List<Try<Void, ? extends ValidationException>> exceptions = new ArrayList<>();
+
+    /* sanity check to ensure uniqueness */
+    exceptions.add(checkUniqueness(gridContainer.allEntitiesAsList()));
```

Comment:
We're checking for duplicates twice: Once here with all entities, then below again with grid elements only, system participants only etc.
I think the problem with this one is that it should actually fail, because the first entity is an `AssetInput`, but there are also entities without id (`GraphicInputs`).

So I think we need to structure validation differently (which also has the benefit of being more user-friendly anyways). My suggestion would be offering various methods for validating asset inputs, result entities, time-based entities etc. (and also a general one for UniqueEntities), and each time the right field set supplier is automatically taken along. 

Sure this produces more methods, but each time it's typesafe and it's clear what is supposed to happen.